src/domain/gpqueue.ts

```diff
@@ -1,21 +1,52 @@
-import { BigInt, Bytes } from '@graphprotocol/graph-ts';
+import { Address, BigInt, Bytes } from '@graphprotocol/graph-ts';
+import { GenesisProtocol } from '../types/GenesisProtocol/GenesisProtocol';
 import { GPQueue } from '../types/schema';
+import { debug } from '../utils';
 
 export function getGPQueue(id: string): GPQueue {
   let gpQueue = GPQueue.load(id) ;
   if (gpQueue == null) {
     gpQueue = new GPQueue(id);
+    gpQueue.votingMachine = null;
   }
   return gpQueue as GPQueue;
 }
 
 export function updateThreshold(dao: string,
+                                gpAddress: Address,
                                 threshold: BigInt,
                                 paramsHash: Bytes,
                                 organizationId: Bytes): void {
   let gpQueue = getGPQueue(organizationId.toHex());
   gpQueue.threshold =  threshold;
   gpQueue.paramsHash = paramsHash;
+
+  if (gpQueue.votingMachine == null) {
+    let gp = GenesisProtocol.bind(gpAddress);
+    let params = gp.parameters(paramsHash);
+    gpQueue.votingMachine = gpAddress;
+    gpQueue.queuedVoteRequiredPercentage = params.value0; // queuedVoteRequiredPercentage
+    gpQueue.queuedVotePeriodLimit = params.value1; // queuedVotePeriodLimit
+    gpQueue.boostedVotePeriodLimit = params.value2; // boostedVotePeriodLimit
+    gpQueue.preBoostedVotePeriodLimit = params.value3; // preBoostedVotePeriodLimit
+    gpQueue.thresholdConst = params.value4; // thresholdConst
+    gpQueue.limitExponentValue = params.value5; // limitExponentValue
+    gpQueue.quietEndingPeriod = params.value6; // quietEndingPeriod
+    gpQueue.proposingRepReward = params.value7;
+    gpQueue.votersReputationLossRatio = params.value8; // votersReputationLossRatio
+    gpQueue.minimumDaoBounty = params.value9; // minimumDaoBounty
+    gpQueue.daoBountyConst = params.value10; // daoBountyConst
+    gpQueue.activationTime = params.value11; // activationTime
+    gpQueue.voteOnBehalf = params.value12; // voteOnBehalf
+  }
   gpQueue.dao = dao;
   gpQueue.save();
 }
+
+export function setScheme(
```

Comment:
why gpque need to have the scheme name ?
do we need this function?
why scheme cannot be passed as a param to updateThreshold ?

Reply:
Because you don't necessarily have the scheme whenever calling updateThreshold

Reply:
for example when ?

Reply:
https://github.com/daostack/subgraph/pull/184/files#diff-b75c7488932d0055e4cddbd7f02c2ed7R179

Reply:
well , in that case you can get the scheme from the proposal it self  as you already reading that at https://github.com/daostack/subgraph/pull/184/files#diff-b75c7488932d0055e4cddbd7f02c2ed7R170 . proposal.callbacks (value01).